### DISCUSSION TITLE
Dependency matching on multiple RIGHT_ATTRS for a token

### DISCUSSION THREAD

**Discussion by Lolologist** [explosion/spaCy#8492] [Help: Coding & Implementations] [answered]
Hi there!
I'm trying to make some matcher patterns with DependencyMatcher. For the most part it's going swimmingly.
I can't seem to figure out how to make a patterns, though, that accomplishes the following:

I want to match an ORG entity type token that _also_ is the nsubjpass or dobj of the ROOT of the sentence.

Like so:

{
        "RIGHT_ID": "root",
        "RIGHT_ATTRS": {"DEP": "ROOT"}
    },
    {
        "LEFT_ID": "root",
        "REL_OP": ">>",
        "RIGHT_ID": "subject",
        "RIGHT_ATTRS": [{"DEP": {"IN": ["nsubjpass","dobj"]}}, {"ENT_TYPE": {"IN": ["ORG", "GPE"]}}]
    },...

I can't have RIGHT_ATTRS as a list though, and there's no AND operator like with IN.

Any thoughts?

**Comment by Lolologist** [comment, marked as answer]
I'm an idiot. Just add them as extra keys into the one dictionary, don't make multiple dictionaries.

**Reply by polm** [reply]
Glad you figured it out! No reason to be hard on yourself, the system has a lot of moving parts and it's easy to be confused.

If you think there's any way we can make this clearer in the docs please let us know.
